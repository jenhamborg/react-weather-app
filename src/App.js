import React from "react";
import styled from "styled-components";
import MainWeather from "./components/views/mainWeather";

import Clouds from "./assets/images/clouds.jpg";

const Container = styled.div`
  width: 100%;
  min-height: 100vh;
  position: relative;
  background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.95));
  color: #ffffff;
  &:before {
    content: "";
    background: url(${Clouds}) no-repeat center center/cover;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: -1;
  }
`;

const App = () => {
  return (
    <Container>
      <MainWeather />
    </Container>
  );
};

export default App;
