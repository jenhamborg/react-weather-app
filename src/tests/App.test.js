import React from "react";
import { axe } from "jest-axe";
import { render, cleanup } from "@testing-library/react";
import App from "../App";

describe("App component", () => {
  beforeAll(() => {
    render(<App />);
  });
  afterEach(cleanup);
  it("should not have basic accessibility issues", async () => {
    const { container } = render(<App />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

});
