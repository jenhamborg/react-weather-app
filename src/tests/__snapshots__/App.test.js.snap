// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App__Container-sc-oucsgb-0 gAfLvA"
>
  <div
    className="styles__MainContainer-sc-izij1-0 iLpuxe"
  >
    <header
      className="styles__MainHeader-sc-izij1-1 jAngzH"
    >
      <h1>
        Weather Now
      </h1>
      <div
        className="searchInput__Input-sc-7ewft-0 kzurIl"
      >
        <div
          className="input-container"
        >
          <input
            aria-label="search by city or zip code"
            data-testid="input"
            label="search by city or zip code"
            maxLength="25"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter City or Zip Code"
            type="search"
            value=""
          />
        </div>
        <div
          className="input-error-message"
        />
      </div>
    </header>
    <div
      className="styles__MainContent-sc-izij1-2 emyXXB"
    >
      <div
        className="main-button-group"
      >
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          New York
        </button>
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          Washington DC
        </button>
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          Chicago
        </button>
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          Minneapolis
        </button>
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          Dallas
        </button>
        <button
          className="mediumButton__Button-sc-1e5cium-0 dScmLB"
          onClick={[Function]}
        >
          Los Angeles
        </button>
      </div>
    </div>
  </div>
</div>
`;
